classes/UpgradeSelfCheck.php

```diff
@@ -401,9 +402,29 @@ private function checkPhpVersionNeedsUpgrade()
      */
     private function checkShopIsDeactivated()
     {
-        return
-            !Configuration::get('PS_SHOP_ENABLE')
-            || (isset($_SERVER['HTTP_HOST']) && in_array($_SERVER['HTTP_HOST'], array('127.0.0.1', 'localhost', '[::1]')));
+        // always return true in localhost
+        if (isset($_SERVER['HTTP_HOST']) && in_array($_SERVER['HTTP_HOST'], array('127.0.0.1', 'localhost', '[::1]'))) {
+            return true;
+        }
```

Comment:
It could be great to improve this check when the port is not 80. If you're using `https://localhost:8080` for instance, this won't work.
<img width="423" alt="Capture d’écran 2022-01-13 à 11 32 21" src="https://user-images.githubusercontent.com/2168836/149313961-bb0e2c6b-4d9d-47e9-be6c-edf53daec676.png">
WDYT?

Reply:
Deep topic 😊 https://stackoverflow.com/questions/2053245/how-can-i-detect-if-the-user-is-on-localhost-in-php

Reply:
It also does not work when using the `NAME.localhost` (default behavior of all modern browsers).
I'm using `https://ps-develop.localhost` and I don't need to configure my `/etc/hosts` to make it works

Reply:
Using `$_SERVER['REMOTE_ADDR']` seems to solve both issue of port and `NAME.localhost`

Reply:
`REMOTE_ADDR` is not working well when you use a reverse proxy engine :)

Reply:
You can look at the `Tools::getRemoteAddr`, from what I remember in one version there was a fix with `HTTP_X_FORWARDED_FOR`. It would be best to just copy this method to the module.